File: Irrgarten_ruby/labyrinth_character.rb
# frozen_string_literal: true

module Irrgarten
  class LabyrinthCharacter

    attr_reader :row, :col, :intelligence, :strength, :health
    attr_writer :health

    def initialize(name,intelligence,strength,health)

      @name = name
      @intelligence = intelligence
      @strength = strength
      @health = health
      @row
      @col
    end

    def initialize_copy(other)

      new(other.name,other.intelligence,other.strength,other.health)
    end

    def dead

      @health < 0
    end

    def set_pos(row,col)

      @row = row
      @col = col
    end

    def to_s

      "Pos:(#{@col},#{@row}) -Nombre: #{@name} -Vida: #{@health} -Fuerza: #{@strength} -Inteligencia: #{@intelligence}"
    end

    def got_wounded

      @health=@health-1
    end
  end
end

File: TestP1/src/irrgarten/LabyrinthCharacter.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author zapi24
 */
public abstract class LabyrinthCharacter {
    
    private String name;
    private float intelligence;
    private float strength;
    private float  health;
    private int row;
    private int col;

    public LabyrinthCharacter(String name, float intelligence, float strength, float health) {
        
        this.name = name;
        this.intelligence = intelligence;
        this.strength = strength;
        this.health = health;
    }
    
    public LabyrinthCharacter(LabyrinthCharacter other){
    
    
        this(other.name,other.intelligence,other.strength,other.health);
    }
    
    public boolean dead(){
        
        return health < 0;
    }

    public int getRow() {
        return row;
    }

    public int getCol() {
        return col;
    }

    protected float getIntelligence() {
        return intelligence;
    }

    protected float getStrength() {
        return strength;
    }

    protected float getHealth() {
        return health;
    }
    
    protected void setHealth(float health){
        
        this.health = health;
    }
    
    public void setPos(int row, int col){
    
        this.row=row;
        this.col=col;
    }
    
    @Override
    public String toString(){
        
        return "Pos: ("+row+","+col+") -Nombre: "+name+" -Vida: "+health+" -Fuerza: "+strength+" -Inteligencia: "+intelligence;
    }
    
    protected void gotWounded(){
        
        this.health--;
    }
    
    abstract float attack();
            
    abstract boolean defend(float attack);  
        
    
    
    
    
    
    
    
}
